sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1225,62 +1225,174 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
         }
 
         /// <summary>
-        ///   Begin processing the requested partition in the background and update tracking state
+        ///   Attempts to begin processing the requested partition in the background and update tracking state
         ///   so that processing can be stopped.
         /// </summary>
         ///
         /// <param name="partitionId">The identifier of the Event Hub partition whose processing should be started.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
-        private async Task StartProcessingPartitionAsync(string partitionId,
-                                                         CancellationToken cancellationToken)
+        /// <returns><c>true</c> if processing was successfully started; otherwise, <c>false</c>.</returns>
+        ///
+        /// <remarks>
+        ///   Exceptions encountered in this method will be logged and will result in the error handler being
+        ///   invoked.  They will not be surfaced to callers.  This is intended to be a safe operation consumed
+        ///   as part of the load balancing cycle, which is failure-tolerant.
+        /// </remarks>
+        ///
+        private async Task<bool> TryStartProcessingPartitionAsync(string partitionId,
```

Comment:
Nothing uses it, but it felt really weird to have a `StartProcessingPartitionAsync` that quite possibly failed silently.   I'll definitely grant you that nothing is keying on the return today, and I don't have plans for it, but I was attempting to provide better semantics with the `TryXXX` pattern so that readers of the code could better grok that it may-or-may not be successful, but we treat it the same either way.

Definitely open to better ideas here.

Reply:
Makes sense.  I was more curious if there was just a subtle "oh shoot - I did mean to use it in this one place" sort of thing going on here.